tests/default/data/pdo-default-fetch-types.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace PdoDefaultFetchTypes;
+
+use PDO;
+use function PHPStan\Testing\assertType;
+
+class HelloWorld
+{
+    public function defaultFetchType(PDO $pdo, string $q): void
+    {
+        $stmt = $pdo->query($q);
+        assertType('PDOStatement<array<bool|float|int|string>>', $stmt);
+        foreach ($stmt as $row) {
+            assertType('array<bool|float|int|string>', $row);
```

Comment:
IMO the PR makes sense, vague types are better than mixed. 

Raw PDO I don't think it can do bool, doctrine I think yes (maybe depending on API used).

Reply:
hmm I might have missed `null` here.. need to double check